hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -78,18 +81,17 @@ public void register(Statement statement, boolean cancelable) {
 	public void release(Statement statement) {
 		log.tracev( "Releasing statement [{0}]", statement );
 
-		// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
-		// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-		if ( log.isDebugEnabled() && !xref.containsKey( statement ) ) {
```

Comment:
The `containsKey` call is only executed when the `DEBUG` level is enabled for ORM loggers.

Are we optimizing the behavior of ORM when using logging at the DEBUG level now? Maybe it's a bit too much... ?